subtree-of-another-tree/Leo.py

```diff
@@ -0,0 +1,29 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
+        def dfs(root, subroot):
+            if not root and not subroot:
+                return True
+            if not root or not subroot:
+                return False
+            if root.val != subroot.val:
+                return False
+
+            return dfs(root.left, subroot.left) and dfs(root.right, subroot.right)
+
+        def solve(root, subroot):
+            if not root:
+                return False
+            if dfs(root, subroot):
+                return True
+            return solve(root.left, subroot) or solve(root.right, subroot)
```

Comment:
nit: `subRoot`는 `solve()`가 아무리 많이 호출되도 바뀌지 않으니, 굳이 매개 변수로 매번 넘길 필요있을까 생각이 들었습니다. (처음에는 `subroot`와 `subRoot`의 미세한 철자 차이를 못 알아봤네요 👀)

```suggestion
        def solve(root):
            if not root:
                return False
            if dfs(root, subRoot):
                return True
            return solve(root.left) or solve(root.right)
```

Reply:
아 그렇네요 수정해서 테스트 해보겠습니다! subroot 도 그냥 쓰면 되는데 뭔가 똑같으면 안된다는 강박이 있어서 바꿨더니 오히려 더 헷갈리게 만드나보네요 😅 그 부분도 수정하겠습니당!